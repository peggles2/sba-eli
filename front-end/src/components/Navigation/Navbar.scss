@import "../../assets/styles/base";

.navbar.ui.menu {
  height: 65px;
  background-color: $dark-blue;

  a {
    padding-bottom: 0;
  }

  .dropdown {
    padding-bottom: 0;
    margin-right: 20px;
  }

  #navbar-learning-paths.active {
    background-color: $grey1;

    .learning-path-item {
      background-color: $grey1 !important;
    }

    .learning-path-item:hover {
      background-color: $light-blue !important;

      .text {
        color: white;
      }
    }
  }

  .dropdown.visible {
    .text {
      color: black;
    }

    img {
      margin-right: 20px;
    }

    .topic-summary {
      margin-left: 15px;

      .path-description {
        white-space: normal;
        margin-top: 15px;
      }
    }
  }


  .search-bar {
    padding: 0;
    padding-top: 13px;

    .fields {
      margin-bottom: 0;

      .search-input {
        padding-right: 0;
      }
    }

    .button {
      background-color: $light-blue;
      color: white;
    }
  }

  .text {
    color: white;
    font-size: 1.2em;
  }

  img.logo {
    width: 2.5em;
  }

  .registrationButtons {
    padding-bottom: 0;

    .button {
      background-color: $dark-blue;
      color: white;
    }
  }

  .userButtons {
    color: white;

    a {
      color: white;
    }

    padding-bottom: 0;
  }

  .greeting {
    padding-bottom: 11px;
  }

  .logoutLink {
    margin-right: 90px;
    cursor: pointer;
  }
}

.ui.button.invisible {
  display: none;
}