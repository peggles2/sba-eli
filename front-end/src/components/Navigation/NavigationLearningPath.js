import React, { Component } from "react";
import { Link } from "react-router-dom";
import { Dropdown, Header } from "semantic-ui-react";
import NavigationLearningObjective from "./NavigationLearningObjective";
import axios from "axios";

export default class NavigationLearningPath extends Component {
  constructor(props) {
    super(props);

    this.state = {
      learningPaths: []
    };
  };

  handleItemClick = (e, {name}) => this.setState({
    activeItem: name
  });

  componentDidMount() {
    this.fetchData();
  }

  fetchData() {
    const url = process.env.REACT_APP_SERVICE_HOST + "/learning_paths";

    axios
      .get(url)
      .then(res => {
        const learningPaths = res.data;
        this.setState({ learningPaths });
      })
      .catch(error => {
        console.error(error);
      });
  }

  render() {
    const {activeItem} = this.state
    
    return this.state.learningPaths.map(lp => (
      <Dropdown.Item key={lp.id} style={{width: '300px'}}>
        <Dropdown pointing='left' fluid text={lp.name}>
          <Dropdown.Menu style={{width: '450px' }}>
            <Header as='h1'><Link to={`/learning_paths/${lp.id}`} onClick={this.handleItemClick}>{lp.name}</Link></Header>
            <NavigationLearningObjective learningPathId={lp.id} />
          </Dropdown.Menu>
        </Dropdown>
      </Dropdown.Item>
    ));
  }
}
