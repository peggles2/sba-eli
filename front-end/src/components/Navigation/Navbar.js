import React, { Component } from 'react';
import {Dropdown, Form, Input, Menu} from 'semantic-ui-react';
import NavigationLearningPath from './NavigationLearningPath';

export default class Navbar extends Component {
  state = {}

  render() {

    return(
      <div className="navbar">
        <Menu>
          <Menu.Item header href={`/`}>Dashboard</Menu.Item>
          <Dropdown text='Learning Paths' item>
            <Dropdown.Menu>
              <NavigationLearningPath />
            </Dropdown.Menu>
          </Dropdown>
          <Menu.Item>
            <Form method='GET' action='/search'>
              <Form.Group inline>
                <Form.Input icon='search' placeholder='Search' name='searchTerm' />
                <Form.Button type="submit">
                  Submit
                </Form.Button>
              </Form.Group>
            </Form>
          </Menu.Item>
        </Menu>
      </div>
    )
  }
};
