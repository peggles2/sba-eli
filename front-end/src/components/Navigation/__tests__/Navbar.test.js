import React from 'react';
import { BrowserRouter as Router,Link } from 'react-router-dom';
import { Button, Menu, Dropdown } from 'semantic-ui-react';
import Navbar from '../Navbar';
import NavigationLearningPath from '../NavigationLearningPath';
import { shallow, mount } from 'enzyme';

describe('Navbar', () => {

  it('should render a <Menu>', () => {
    const wrapper = shallow(<Navbar/>);

    expect(wrapper.find(Menu).length).toEqual(1);
    expect(wrapper.find(Menu.Menu).exists()).toBe(true);
  });

  it('should render a <Dropdown> of all learning paths', () => {
    const wrapper = shallow(<Navbar/>);

    expect(wrapper.find(Dropdown).length).toEqual(1);
    expect(wrapper.find(Dropdown.Menu).length).toEqual(1);
    expect(wrapper.find(NavigationLearningPath).length).toEqual(1);
  });

  it('should render a Login button', () => {
    const wrapper = mount(<Router><Navbar/></Router>);

    expect(wrapper.find(Link).exists()).toBe(true);
  });

  it('should render a Register button', () => {
    const wrapper = mount(<Router><Navbar/></Router>);

    expect(wrapper.find(Link).exists()).toBe(true);
  });
});
