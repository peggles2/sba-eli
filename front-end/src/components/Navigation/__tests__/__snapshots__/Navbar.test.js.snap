// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Menu>
        <MenuItem
          header={true}
          href="/"
        >
          Dashboard
        </MenuItem>
        <Dropdown
          additionLabel="Add "
          additionPosition="top"
          closeOnBlur={true}
          deburr={false}
          icon="dropdown"
          item={true}
          minCharacters={1}
          noResultsMessage="No results found."
          openOnFocus={true}
          renderLabel={[Function]}
          searchInput="text"
          selectOnBlur={true}
          selectOnNavigation={true}
          text="Learning Paths"
          wrapSelection={true}
        >
          <DropdownMenu>
            <NavigationLearningPath />
          </DropdownMenu>
        </Dropdown>
        <MenuItem>
          <Form
            action="/search"
            as="form"
            method="GET"
          >
            <FormGroup
              inline={true}
            >
              <FormInput
                as={[Function]}
                control={[Function]}
                icon="search"
                name="searchTerm"
                placeholder="Search"
              />
              <FormButton
                as={[Function]}
                control={[Function]}
                type="submit"
              >
                Submit
              </FormButton>
            </FormGroup>
          </Form>
        </MenuItem>
      </Menu>,
      "className": "navbar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <MenuItem
            header={true}
            href="/"
          >
            Dashboard
          </MenuItem>,
          <Dropdown
            additionLabel="Add "
            additionPosition="top"
            closeOnBlur={true}
            deburr={false}
            icon="dropdown"
            item={true}
            minCharacters={1}
            noResultsMessage="No results found."
            openOnFocus={true}
            renderLabel={[Function]}
            searchInput="text"
            selectOnBlur={true}
            selectOnNavigation={true}
            text="Learning Paths"
            wrapSelection={true}
          >
            <DropdownMenu>
              <NavigationLearningPath />
            </DropdownMenu>
          </Dropdown>,
          <MenuItem>
            <Form
              action="/search"
              as="form"
              method="GET"
            >
              <FormGroup
                inline={true}
              >
                <FormInput
                  as={[Function]}
                  control={[Function]}
                  icon="search"
                  name="searchTerm"
                  placeholder="Search"
                />
                <FormButton
                  as={[Function]}
                  control={[Function]}
                  type="submit"
                >
                  Submit
                </FormButton>
              </FormGroup>
            </Form>
          </MenuItem>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Dashboard",
            "header": true,
            "href": "/",
          },
          "ref": null,
          "rendered": "Dashboard",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "additionLabel": "Add ",
            "additionPosition": "top",
            "children": <DropdownMenu>
              <NavigationLearningPath />
            </DropdownMenu>,
            "closeOnBlur": true,
            "deburr": false,
            "icon": "dropdown",
            "item": true,
            "minCharacters": 1,
            "noResultsMessage": "No results found.",
            "openOnFocus": true,
            "renderLabel": [Function],
            "searchInput": "text",
            "selectOnBlur": true,
            "selectOnNavigation": true,
            "text": "Learning Paths",
            "wrapSelection": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <NavigationLearningPath />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Form
              action="/search"
              as="form"
              method="GET"
            >
              <FormGroup
                inline={true}
              >
                <FormInput
                  as={[Function]}
                  control={[Function]}
                  icon="search"
                  name="searchTerm"
                  placeholder="Search"
                />
                <FormButton
                  as={[Function]}
                  control={[Function]}
                  type="submit"
                >
                  Submit
                </FormButton>
              </FormGroup>
            </Form>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "action": "/search",
              "as": "form",
              "children": <FormGroup
                inline={true}
              >
                <FormInput
                  as={[Function]}
                  control={[Function]}
                  icon="search"
                  name="searchTerm"
                  placeholder="Search"
                />
                <FormButton
                  as={[Function]}
                  control={[Function]}
                  type="submit"
                >
                  Submit
                </FormButton>
              </FormGroup>,
              "method": "GET",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <FormInput
                    as={[Function]}
                    control={[Function]}
                    icon="search"
                    name="searchTerm"
                    placeholder="Search"
                  />,
                  <FormButton
                    as={[Function]}
                    control={[Function]}
                    type="submit"
                  >
                    Submit
                  </FormButton>,
                ],
                "inline": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": [Function],
                    "control": [Function],
                    "icon": "search",
                    "name": "searchTerm",
                    "placeholder": "Search",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": [Function],
                    "children": "Submit",
                    "control": [Function],
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Submit",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Menu>
          <MenuItem
            header={true}
            href="/"
          >
            Dashboard
          </MenuItem>
          <Dropdown
            additionLabel="Add "
            additionPosition="top"
            closeOnBlur={true}
            deburr={false}
            icon="dropdown"
            item={true}
            minCharacters={1}
            noResultsMessage="No results found."
            openOnFocus={true}
            renderLabel={[Function]}
            searchInput="text"
            selectOnBlur={true}
            selectOnNavigation={true}
            text="Learning Paths"
            wrapSelection={true}
          >
            <DropdownMenu>
              <NavigationLearningPath />
            </DropdownMenu>
          </Dropdown>
          <MenuItem>
            <Form
              action="/search"
              as="form"
              method="GET"
            >
              <FormGroup
                inline={true}
              >
                <FormInput
                  as={[Function]}
                  control={[Function]}
                  icon="search"
                  name="searchTerm"
                  placeholder="Search"
                />
                <FormButton
                  as={[Function]}
                  control={[Function]}
                  type="submit"
                >
                  Submit
                </FormButton>
              </FormGroup>
            </Form>
          </MenuItem>
        </Menu>,
        "className": "navbar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <MenuItem
              header={true}
              href="/"
            >
              Dashboard
            </MenuItem>,
            <Dropdown
              additionLabel="Add "
              additionPosition="top"
              closeOnBlur={true}
              deburr={false}
              icon="dropdown"
              item={true}
              minCharacters={1}
              noResultsMessage="No results found."
              openOnFocus={true}
              renderLabel={[Function]}
              searchInput="text"
              selectOnBlur={true}
              selectOnNavigation={true}
              text="Learning Paths"
              wrapSelection={true}
            >
              <DropdownMenu>
                <NavigationLearningPath />
              </DropdownMenu>
            </Dropdown>,
            <MenuItem>
              <Form
                action="/search"
                as="form"
                method="GET"
              >
                <FormGroup
                  inline={true}
                >
                  <FormInput
                    as={[Function]}
                    control={[Function]}
                    icon="search"
                    name="searchTerm"
                    placeholder="Search"
                  />
                  <FormButton
                    as={[Function]}
                    control={[Function]}
                    type="submit"
                  >
                    Submit
                  </FormButton>
                </FormGroup>
              </Form>
            </MenuItem>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Dashboard",
              "header": true,
              "href": "/",
            },
            "ref": null,
            "rendered": "Dashboard",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "additionLabel": "Add ",
              "additionPosition": "top",
              "children": <DropdownMenu>
                <NavigationLearningPath />
              </DropdownMenu>,
              "closeOnBlur": true,
              "deburr": false,
              "icon": "dropdown",
              "item": true,
              "minCharacters": 1,
              "noResultsMessage": "No results found.",
              "openOnFocus": true,
              "renderLabel": [Function],
              "searchInput": "text",
              "selectOnBlur": true,
              "selectOnNavigation": true,
              "text": "Learning Paths",
              "wrapSelection": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <NavigationLearningPath />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Form
                action="/search"
                as="form"
                method="GET"
              >
                <FormGroup
                  inline={true}
                >
                  <FormInput
                    as={[Function]}
                    control={[Function]}
                    icon="search"
                    name="searchTerm"
                    placeholder="Search"
                  />
                  <FormButton
                    as={[Function]}
                    control={[Function]}
                    type="submit"
                  >
                    Submit
                  </FormButton>
                </FormGroup>
              </Form>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "action": "/search",
                "as": "form",
                "children": <FormGroup
                  inline={true}
                >
                  <FormInput
                    as={[Function]}
                    control={[Function]}
                    icon="search"
                    name="searchTerm"
                    placeholder="Search"
                  />
                  <FormButton
                    as={[Function]}
                    control={[Function]}
                    type="submit"
                  >
                    Submit
                  </FormButton>
                </FormGroup>,
                "method": "GET",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <FormInput
                      as={[Function]}
                      control={[Function]}
                      icon="search"
                      name="searchTerm"
                      placeholder="Search"
                    />,
                    <FormButton
                      as={[Function]}
                      control={[Function]}
                      type="submit"
                    >
                      Submit
                    </FormButton>,
                  ],
                  "inline": true,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": [Function],
                      "control": [Function],
                      "icon": "search",
                      "name": "searchTerm",
                      "placeholder": "Search",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": [Function],
                      "children": "Submit",
                      "control": [Function],
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": "Submit",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
